packages/tools/src/tools/CursorCrosshairSync.ts

```diff
@@ -0,0 +1,581 @@
+    this._currentCanvasPosition = currentPoints.canvas;
+
+    const annotation = this.getActiveAnnotation(element);
+    if (annotation === null) {
+      this.addNewAnnotation(evt);
+      return false;
+    }
+    this.updateAnnotationPosition(element, annotation);
+    return false;
+  };
+
+  //image change event seems to fire before image is rendered, so in order to get correct world position we need to wait for next render event
+  handleImageChange = (
+    evt:
+      | Types.EventTypes.StackViewportScrollEvent
+      | Types.EventTypes.VolumeNewImageEvent
+  ): void => {
+    const element = evt.target as HTMLDivElement;
+    element.addEventListener(
+      Enums.Events.IMAGE_RENDERED,
+      () => {
+        if (!this) return;
+        const viewport = getEnabledElement(element)?.viewport;
+        if (!viewport) return;
+        const renderingEngine = viewport.getRenderingEngine();
+        if (!renderingEngine) return;
+
+        //calculate new world position from chached canvas position
+        const activeAnnotation = this.getActiveAnnotation(element);
+
+        if (!this._currentCanvasPosition || !activeAnnotation) return;
+        const worldPos = viewport.canvasToWorld(this._currentCanvasPosition);
+        this._currentMouseWorldPosition = worldPos;
+
+        this.updateAnnotationPosition(element, activeAnnotation);
+      },
+      {
+        once: true,
+      }
+    );
+  };
+
+  //when the mouse is moved over a div, attach an event listener to this div to update the world position of the annotation when stack is scrolled
+  setMouseOverElement(element: HTMLDivElement | null): void {
+    if (element === this._mouseOverElement) return;
+    const previousElement = this._mouseOverElement;
+    this._mouseOverElement = element;
+    if (previousElement) {
+      previousElement.removeEventListener(
+        Enums.Events.VOLUME_NEW_IMAGE,
+        this.handleImageChange as EventListener
+      );
+      previousElement.removeEventListener(
+        Enums.Events.STACK_VIEWPORT_SCROLL,
+        this.handleImageChange as EventListener
+      );
+    }
+    if (element) {
+      this._mouseOverElement = element;
+      element.addEventListener(
+        Enums.Events.VOLUME_NEW_IMAGE,
+        this.handleImageChange as EventListener
+      );
+      element.addEventListener(
+        Enums.Events.STACK_VIEWPORT_SCROLL,
+        this.handleImageChange as EventListener
+      );
+    }
+  }
+
+  getActiveAnnotation(element: HTMLDivElement): null | Annotation {
+    const annotations = getAnnotations(element, this.getToolName());
+    if (annotations === undefined || annotations.length === 0) {
+      return null;
+    }
+    const targetAnnotation = annotations[0];
+    if (annotations.length > 1) {
+      const wrongAnnotations = annotations.slice(1);
+      wrongAnnotations.forEach((annotation) => {
+        if (annotation.annotationUID)
+          removeAnnotation(annotation.annotationUID, element);
```

Comment:
I guess you have to removeAnnotation since each click will create a new one (via addNewAnnotation)?

Reply:
Do we want to add a early check on `addAnnotation` that check if there is one annotation, return early so that we don't add annotation on each click?